### ISSUE TITLE
[BUG] private package installation fails because habitat token is not used when determining candidate version

### ISSUE THREAD

**Issue by st-h** [chef-cookbooks/habitat#102]
### Cookbook version
0.56.1, 0.57.0

### Chef-client version
14.2.0

### Platform Details
not applicable

### Scenario:
I am trying to set up habitat supervisor and services using this chef cookbook. All seems to work fine if I try to install `hab_package` one of the `core/*` packages. However, if I try to install a private package I have created myself and uploaded to builder using `hab pkg upload`, the chef run fails with the message:
```
Chef::Exceptions::Package
-------------------------
No candidate version available for private_origin/mypackage
``` 

### Steps to Reproduce:
- upload package to a private origin (public builder is fine)
- make use of this cookbook to install package via `hab_package` resource

chef recipe, fails with error from above:
```
ENV['HAB_AUTH_TOKEN'] = '_mytoken'

hab_sup 'default' do
  action :run
end

hab_package 'private_origin/mypackage' do
  action :install
end
```

when using service instead of package:

```
hab_service 'private_origin/mypackage' do
  action :load
  strategy 'at-once'
end
```
it fails with the following output:
```
execute[hab svc load private_origin/mypackage --channel stable --strategy at-once] (/tmp/kitchen/cache/cookbooks/habitat/resources/service.rb line 93) had an error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0], but received '1'
---- Begin output of hab svc load private_origin/mypackage --channel stable --strategy at-once ----
STDOUT: » Installing private_origin/mypackage
☁ Determining latest version of private_origin/mypackage in the 'stable' channel
STDERR: ✗✗✗
✗✗✗ [Err: 0] hab-sup(ER)[components/sup/src/error.rs:455:9]: Package not found. If you are attempting to install from a local depot with an upstream configured, try again in a few seconds.
✗✗✗
```
### Expected Result:
There is a way to provide the habitat token or any other means to authenticate at builder, in order to being able to download private packages.

### Actual Result:
- the resource does not provide any way to provide a token
- adding `ENV['HAB_AUTH_TOKEN'] = '_mytoken'` to my cookbook prior to calling `hab_package` and `hab_sup` does not work.
- However, habitat specifies that `HAB_AUTH_TOKEN` should be set as an environment variable. I could confirm that this works when I log into the node, stop the supervisor, `export HAB_AUTH_TOKEN='_token'` and start the supervisor again. So, either I am setting the `HAB_AUTH_TOKEN` incorrectly within the chef recipe, or for some reason it does not get picked up.
 
I found that it even is possible to print out the token via 
```
execute 'print hab token' do
  command '(>&2 echo "token=$HAB_AUTH_TOKEN")'
  live_stream true
end
``` 

prior to installing the supervisor, however the token does not get picked up by the supervisor that gets installed. So probably this is much more related to how the supervisor is installed and started, than a way to set the token.

**Comment by st-h** [comment]
After looking through the code and making various attempts to work around this, I think we at least need a way to provide the token prior to starting the supervisor via upstart, systemd etc... Maybe that will even fix the issue with initial installation as well.

**Comment by st-h** [comment]
I was able to put some more time into this, however the only thing I could find that improves the situation is modifying the (systemd) template.

```
       systemd_unit "hab-sup-#{new_resource.override_name}.service" do
          content(Unit: {
                    Description: 'The Habitat Supervisor',
                  },
                  Service: {
                    ExecStart: "/bin/hab sup run #{exec_start_options}",
                    Restart: 'on-failure',
                    Environment: 'HAB_AUTH_TOKEN=_HABTOKEN'
                  },
                  Install: {
                    WantedBy: 'default.target',
                  })
          action :create
        end
```
Now, hab is able to deal with private packages/origin when issues svc or sup related commands. Good thing is this also works across reboots. However, pkg related commands do still fail. The only workaround I have found so far is to explicitly provide the hab token when running those commands.

It would be really helpful if we could get a response from the maintainers on:
- if they had any intention how dealing with private packages should work
- if not, how we should progress with this issue

I can put together a PR trying to make it work the way I mentioned above. However, I don't think it makes much sense to implement stuff when it is not even clear if there were any intention on how this feature should work in first place. So, any response is highly appreciated.

**Comment by st-h** [comment]
I think I finally got to the root cause of this, which is related to a failure to pick up the hab token - even though a little different than what I suspected so far.

The issue is that `provider_hab_package.rb` bypasses the hab client when trying to figure out the candidate version. It builds a url to check if the package is present on builder:
https://github.com/chef-cookbooks/habitat/blob/2dca4601b491949dcf8d9595a32a0577e0f60a1a/libraries/provider_hab_package.rb#L117
and then uses a plain http request. Therefore it never picks up the hab token:
https://github.com/chef-cookbooks/habitat/blob/2dca4601b491949dcf8d9595a32a0577e0f60a1a/libraries/provider_hab_package.rb#L119

Therefore private packages appear to not exist to the hab_package provider and therefore do not get installed, leading to a `No candidate version available for xyz` exception.

I see two options to fixing this:
- use the hab client to search for the package
- provide the `Authorization` Header when invoking the http client, so the token gets picked up
